marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -56,11 +58,42 @@ function getArtifacts(artifactType) {
         queriesForAnd.push(cts.collectionQuery(coll));
     }
     if (queriesForAnd.length) {
-        const results = cts.search(queriesForAnd.length > 1 ? cts.andQuery(queriesForAnd) : queriesForAnd[0]);
-        for (const result of results) {
-            artifacts.push(result.toObject());
+        if (artifactType == "loadData") {
+            const results = cts.search(queriesForAnd.length > 1 ? cts.andQuery(queriesForAnd) : queriesForAnd[0]);
+            for (const result of results) {
+                artifacts.push(result.toObject());
+            }
+            return artifacts;
+        } else {
+            return getArtifactsGroupbyEntity(artifactType, queriesForAnd)
         }
     }
+    return [];
+}
+
+function getArtifactsGroupbyEntity(artifactType, queriesForAnd) {
+    const artifacts = [];
+    const entityNames = getEntityNames();
+    for (const ename of entityNames) {
+        let configByEntity = {};
+        configByEntity.name = ename;
+        queriesForAnd.push(cts.jsonPropertyValueQuery("entityName", ename));
+        let configs = cts.search(cts.andQuery(queriesForAnd));
+        // artifactType can be mapping, mastering, merging
+        switch (artifactType) {
+            case "mapping":
```

Comment:
Ideally, this service wouldn't have much knowledge of the concrete artifact types. Is it necessary to store the "configs" under different key names? Couldn't it just be "config" or "settings", regardless of the artifact type?

Reply:
agree. we can use config. @ryanjdew do you agree?